lib/api/exception/network_exceptions.dart

```diff
@@ -0,0 +1,158 @@
+import 'dart:io';
+
+import 'package:dio/dio.dart';
+import 'package:freezed_annotation/freezed_annotation.dart';
+
+part 'network_exceptions.freezed.dart';
+
+@freezed
+class NetworkExceptions with _$NetworkExceptions {
+  const factory NetworkExceptions.requestCancelled() = RequestCancelled;
+
+  const factory NetworkExceptions.unauthorisedRequest() = UnauthorisedRequest;
+
+  const factory NetworkExceptions.badRequest() = BadRequest;
+
+  const factory NetworkExceptions.notFound(String reason) = NotFound;
+
+  const factory NetworkExceptions.methodNotAllowed() = MethodNotAllowed;
+
+  const factory NetworkExceptions.notAcceptable() = NotAcceptable;
+
+  const factory NetworkExceptions.requestTimeout() = RequestTimeout;
+
+  const factory NetworkExceptions.sendTimeout() = SendTimeout;
+
+  const factory NetworkExceptions.conflict() = Conflict;
+
+  const factory NetworkExceptions.internalServerError() = InternalServerError;
+
+  const factory NetworkExceptions.notImplemented() = NotImplemented;
+
+  const factory NetworkExceptions.serviceUnavailable() = ServiceUnavailable;
+
+  const factory NetworkExceptions.noInternetConnection() = NoInternetConnection;
+
+  const factory NetworkExceptions.formatException() = FormatException;
+
+  const factory NetworkExceptions.unableToProcess() = UnableToProcess;
+
+  const factory NetworkExceptions.defaultError(String error) = DefaultError;
+
+  const factory NetworkExceptions.unexpectedError() = UnexpectedError;
+
+  static NetworkExceptions fromDioException(error) {
+    if (error is Exception) {
+      try {
+        NetworkExceptions networkExceptions;
+        if (error is DioError) {
+          switch (error.type) {
+            case DioErrorType.cancel:
+              networkExceptions = NetworkExceptions.requestCancelled();
+              break;
+            case DioErrorType.connectTimeout:
+              networkExceptions = NetworkExceptions.requestTimeout();
+              break;
+            case DioErrorType.other:
+              networkExceptions = NetworkExceptions.noInternetConnection();
+              break;
+            case DioErrorType.receiveTimeout:
+              networkExceptions = NetworkExceptions.sendTimeout();
+              break;
+            case DioErrorType.sendTimeout:
+              networkExceptions = NetworkExceptions.sendTimeout();
+              break;
+            case DioErrorType.response:
+              switch (error.response?.statusCode) {
+                case 400:
+                  networkExceptions = NetworkExceptions.unauthorisedRequest();
```

Comment:
badRequest() 🤣

Reply:
you suck

Reply:
Fixed https://github.com/luongvo/flutter-survey/pull/50/commits/0464ce8e3f93e91170a8d1bf1b484d1e55ff9a24